lib/tasks/upgrade.rake

```diff
@@ -0,0 +1,39 @@
+# TRANSLATORS: do not translate
+desc <<-END_DESC
+  This task runs predefined upgrade steps.
+
+  Examples:
+    rake upgrade:run
+    rake upgrade:mark_as_ran TASK_NAME=some_rake_task,another_rake_task
+
+END_DESC
+
+namespace :upgrade do
+  task :run => :environment do
+    tasks = Foreman::Plugin.registered_plugins.values.map{|plugin| plugin.upgrade_tasks}.flatten
+    tasks.select!{|task| task.run?}
+
+    tasks.each_with_index do |task, index|
+      count = "#{index + 1}/#{tasks.count}"
+
+      message = if task.long_running?
+                  _("Upgrade Step %{count}: %{name}, This may take a long while.")
+                else
+                  _("Upgrade Step %{count}: %{name}")
+                end
+
+      puts '============================================='
+      puts message % {:name => task.name, :count => count}
+
+      Rake::Task[task.name].execute
```

Comment:
One thing missing is the ability to pass argument of some sort to these tasks.  Some of the katello ones do use a COMMIT=true environment variable as arguments.  Couple of options:

1.  Continue using ENV variables and adjust the ENV variable prior to calling those tasks (the env variables would be defined in the plugin api)
2.  Use rake task arguments.  I really don't like these, but it has advantages in that its not messing with the ENV variable.  

Thoughts?

Reply:
We could standardize on exporting an env variable (`FOREMAN_UPGRADE=1` for example) so rake tasks can test for that.